visual-java/src/main/java/com/saucelabs/visual/model/FullPageScreenshotConfig.java

```diff
@@ -6,29 +6,68 @@
 public class FullPageScreenshotConfig {
 
   private int delayAfterScrollMs;
+  private Boolean disableCSSAnimation;
   private List<String> hideAfterFirstScroll;
+  private Boolean hideScrollBars;
+  private int scrollLimit;
 
-  public FullPageScreenshotConfig(int delayAfterScrollMs, List<String> hideAfterFirstScroll) {
+  public FullPageScreenshotConfig(
+      int delayAfterScrollMs,
+      Boolean disableCSSAnimation,
+      List<String> hideAfterFirstScroll,
+      Boolean hideScrollBars,
+      int scrollLimit) {
     this.delayAfterScrollMs = delayAfterScrollMs;
+    this.disableCSSAnimation = disableCSSAnimation;
     this.hideAfterFirstScroll = hideAfterFirstScroll;
+    this.hideScrollBars = hideScrollBars;
+    this.scrollLimit = scrollLimit;
   }
 
   public static class Builder {
     private int delayAfterScrollMs;
+    private Boolean disableCSSAnimation;
     private List<String> hideAfterFirstScroll;
+    private Boolean hideScrollBars;
+    private int scrollLimit;
 
     public Builder withDelayAfterScrollMs(int delayAfterScrollMs) {
       this.delayAfterScrollMs = delayAfterScrollMs;
       return this;
     }
 
+    public Builder withDisableCSSAnimation(Boolean disableCSSAnimation) {
+      this.disableCSSAnimation = disableCSSAnimation;
+      return this;
+    }
+
     public Builder withHideAfterFirstScroll(String... hideAfterFirstScroll) {
       this.hideAfterFirstScroll = Arrays.asList(hideAfterFirstScroll);
       return this;
     }
 
+    public Builder withHideScrollBars(Boolean hideScrollBars) {
+      this.hideScrollBars = hideScrollBars;
+      return this;
+    }
+
+    public Builder withScrollLimit(int scrollLimit) {
+      if (scrollLimit > 0 && scrollLimit <= 10) {
+        this.scrollLimit = scrollLimit;
+      } else {
```

Comment:
This should be enforced only by the backend.

If for some reason we raise the scrollLimit, we will have to update the client which is unnecessary burden.

Client should only check that this is a positive value.